spec/light-client/supervisor/supervisor_001_draft.md

```diff
@@ -463,69 +463,94 @@ func Sequential-Supervisor (initdata LCInitData) (Error) {
+trusted period because it already trusts it, however, if the light block is 
+outside the trust period, there is a higher chance the light client won't be
+able to verify anything. 
+
+Cross-checking this trusted block with providers upon initialization is helpful
+for ensuring liveness and correct configuration of such nodes but does not 
+increase trust since proving a conflicting block is a 
+[light client attack](https://informal.systems) 
+and not just a [bogus](https://informal.systems) block could result in
+performing backwards verification beyond the trusted period, thus a fruitless
+endeavour.
+
+However, with the notion of it's better to fail earlier than later, the golang
+light client implementation will perform a consistency check on all providers
+and will error if one returns a different header, allowing the user
+the opportunity to reinitialize.
 
 #### **[LC-FUNC-INIT.1]:**
 
 ```go
-func InitLightClient (initData LCInitData) (LightStore, Error) {
+func InitLightClient(initData LCInitData) (LightStore, Error) {
+    var initialBlock LightBlock
 
-    if LCInitData.LightBlock != nil {
-        // we trust the provided initial block.
-        newblock := LCInitData.LightBlock
-    }
-    else {
-        genesisBlock := makeblock(initData.genesisDoc);
+    switch {
+    case LCInitData.TrustedBlock != nil:
+        // we trust the block from a prior initialization
+        initialBlock = LCInitData.TrustedBlock
 
-        result := NoResult;
-        while result != ResultSuccess {
-            current = FetchLightBlock(PeerList.primary(), genesisBlock.Header.Height + 1)
-            // QUESTION: is the height with "+1" OK?
+    case LCInitData.TrustedHash != nil:
+        untrustedBlock := FetchLightBlock(PeerList.Primary(), LCInitData.TrustedHeight)
+        
 
-            if CANNOT_VERIFY = ValidAndVerify(genesisBlock, current) {
-                Replace_Primary();
-            }
-            else {
-                result = ResultSuccess
-            }
-        }
-  
-        // cross-check
-  auxLS := new LightStore
-  auxLS.Add(current)
-        Evidences := AttackDetector(genesisBlock, auxLS)
-        if Evidences.Empty {
-            newBlock := current
+        // verify that the hashes match
+        if untrustedBlock.Hash() != LCInitData.TrustedHash {
+            return nil, Error("Primary returned block with different hash")
         }
-        else {
-            // [LC-SUMBIT-EVIDENCE.1]
-            submitEvidence(Evidences);
-            return(nil, ErrorAttack);
+        // after checking the hash we now trust the block
+        initialBlock = untrustedBlock        
+    }
+    case LCInitData.Genesis != nil:
+        untrustedBlock := FetchLightBlock(PeerList.Primary(), LCInitData.Genesis.InitialHeight)
+        
+        // verify that 2/3+ of the validator set signed the untrustedBlock
+        if err := VerifyCommit(untrustedBlock.Commit, LCInitData.Genesis.Validators); err != nil {
```

Comment:
Is the function defined somewhere? Previously we used
[`ValidandVerified`](https://github.com/tendermint/spec/blob/master/spec/light-client/verification/verification_002_draft.md#lcv-func-valid2) here (with a typo). As it takes as inputs two lightblocks we generated a gensisblock.

Reply:
It's mentioned here: https://github.com/tendermint/spec/blob/master/spec/light-client/verification/README.md and is the name of a function within the golang codebase. 

AFAICS: `ValidandVerified` does three things
- Check that all formats are consistent and as expected
- Checks that their is a sufficient degree of overlap in the validators between two heights
- Checks that the 2/3+ of the validator set and the untrusted height signed the untrusted block

`VerifyCommit` performs only the last of these three things

Reply:
I see. I think we should add the function definition of `VerifyCommit` to the spec. (Also the linked README.md only contains `VerifyCommitTrusting` and `VerifyCommitFull`)

Reply:
In this case, I'm talking about `VerifyCommitFull`. `VerifyCommitTrusting` is the name of the function that performs the second bullet point

Reply:
Should I add these functions to the verification document?

Reply:
Yes, please

Reply:
Added `VerifyCommitLight` here: https://github.com/tendermint/spec/blob/callum/light-client/spec/light-client/verification/verification_001_published.md#lcv-func-verifycommitlight1

Reply:
It would be nice to have all these verification function well documented in the spec: https://github.com/tendermint/tendermint/blob/master/types/validation.go